scalameta/parsers/shared/src/main/scala/scala/meta/internal/parsers/ScalametaParser.scala

```diff
@@ -513,23 +562,30 @@ class ScalametaParser(input: Input, dialect: Dialect) { parser =>
     def unapply(token: Token): Boolean = {
       token.is[Ident] || token.is[KwSuper] || token.is[KwThis] ||
       token.is[LeftParen] || token.is[At] || token.is[Underscore] ||
-      token.is[Unquote] || (token.is[Literal] && dialect.allowLiteralTypes)
+      token.is[Unquote] || (token.is[Literal] && dialect.allowLiteralTypes) ||
+      token.is[MacroSplicedIdent] || token.is[MacroQuotedIdent]
     }
   }
 
   @classifier
   trait ExprIntro {
     def unapply(token: Token): Boolean = {
-      (token.is[Ident] && !(token.syntax == "inline" && !dialect.allowInlineIdents)) ||
+      (token.is[Ident] && !inlineDefOrOpaque(token)) ||
       token.is[Literal] || token.is[Interpolation.Id] || token.is[Xml.Start] ||
       token.is[KwDo] || token.is[KwFor] || token.is[KwIf] ||
       token.is[KwNew] || token.is[KwReturn] || token.is[KwSuper] ||
       token.is[KwThis] || token.is[KwThrow] || token.is[KwTry] || token.is[KwWhile] ||
       token.is[LeftParen] || token.is[LeftBrace] || token.is[Underscore] ||
-      token.is[Unquote]
+      token.is[Unquote] || token.is[MacroSplice] || token.is[MacroQuote] ||
+      token.is[MacroQuotedIdent] || token.is[MacroSplicedIdent]
     }
   }
 
+  private def inlineDefOrOpaque(token: Token): Boolean = {
+    (token.text == "inline" && (DclIntro.unapply(token) || Modifier.unapply(token))) ||
```

Comment:
it should actually be `token.next`, my bad!

Reply:
are you sure?

Reply:
I think so we are trying to make sure that inline is in a proper place, no? Before a modifier or dclintro. We already know that `token` is inline.